wiki/osu!_File_Formats/Osu_(file_format)/en.md

```diff
@@ -194,26 +194,58 @@ Hit Circle Syntax:
 `x,y,time,type,hitSound,addition`
 `164,260,2434,1,0,0:0:0:0:`
 
+x (Integer) ranges from 0 to 512 (inclusive) and y (Integer) ranges from 0 to 384 (inclusive).
+
+time (Integer) is in milliseconds from the beginning of the song.
+
+type (Integer) is a bitmap:
+
+1 = Circle, 2 = Slider, 4 = New combo, 8 = Spinner. More significant bits can be used to choose the next combo color.
+
+hitSound (Integer) is a bitmap:
+
+2 = Whistle, 4 = Finish, 8 = Clap
+
+The only actual "types" are circles, sliders, and spinners. New combo can be OR'd in to extend the behavior of the hit circle.
+
+addition (Integer:Integer:Integer:Integer) is optional, and defaults to "0:0:0:0:". The first value corresponds to the sample set, and the second value corresponds to additions.
+
+0 = Auto, 1 = Normal, 2 = Soft, 3 = Drum
+
+The third value corresponds to the sample index, e.g. normal-3. The fourth value declares the volume of the sample.
```

Comment:
There's a fifth value that defaults to "", indicating the filename of a custom hitsound, if any were used. They're only really common in osu!mania, but they do exist and work (partially) for other modes